main.go

```diff
@@ -563,9 +573,9 @@ func (c *controller) sync(ctx context.Context) {
 		}
 
 		// If there's an increase in replicas we poll for the new replicas to be ready
-		if _, ok := c.replicas[hashring]; ok && c.replicas[hashring] < *sts.Spec.Replicas {
+		if _, ok := c.replicas[sts.Name]; ok && c.replicas[sts.Name] < *sts.Spec.Replicas {
```

Comment:
found a bug when multiple sts pointing to same replicas, not sure whats the point of using hashring name as the key